tooling/phd/cli/cluster_create.py

```diff
@@ -0,0 +1,213 @@
+) -> None:
+    """
+    Create a new cluster configuration using cookiecutter.
+
+    Args:
+        cluster_name: Name of the cluster (e.g., "PHD Production Cluster")
+        cluster_domain: Domain for the cluster (e.g., "cluster.example.com")
+        environment: Environment name (default: "production")
+        short_description: Short description of the cluster
+        cloud_provider: Cloud provider (aws or digitalocean)
+        harmony_module_version: Harmony module version/commit hash
+        opencraft_module_version: OpenCraft module version
+        picasso_version: Picasso version
+        template_version: PHD cluster template version
+        git_organization: Git organization name
+        git_repository: Git repository URL (if not provided, will be auto-generated)
+        template_repository: Git URL of the cluster template repository
+        output_dir: Directory where cluster config will be created
+
+    Raises:
+        Exception: If cookiecutter fails
+    """
+
+    logger.info("Creating cluster configuration for '%s'", cluster_name)
+
+    cluster_slug = cluster_name.lower().replace(" ", "-").replace("_", "-")
+
+    if not git_repository:
+        git_repository = f"https://github.com/{git_organization}/phd-{cluster_slug}.git"
+
+    output_path = Path(output_dir).resolve()
+    output_path.mkdir(parents=True, exist_ok=True)
+
+    # Use absolute path for local templates
+    if template_repository.startswith((".", "/")):
+        template_path = Path(template_repository).resolve()
+        if template_path.is_dir():
+            template_repository = str(template_path)
+        elif template_path.name == "phd-cluster-template":
+            cluster_template_path = template_path / "cluster-template"
+            if cluster_template_path.is_dir():
+                template_repository = str(cluster_template_path)
+
+    logger.info("generate cluster configuration")
+    logger.info("Using template repository: %s", template_repository)
+
+    extra_context = {
+        "environment": environment,
+        "cluster_name": cluster_name,
+        "cluster_slug": cluster_slug,
+        "cluster_domain": cluster_domain,
+        "short_description": short_description,
+        "cloud_provider": cloud_provider,
+        "harmony_module_version": harmony_module_version,
+        "opencraft_module_version": opencraft_module_version,
+        "picasso_version": picasso_version,
+        "phd_cluster_template_version": template_version,
+        "git_organization": git_organization,
+        "git_repository": git_repository,
+    }
+
+    logger.debug("Template repository: %s", template_repository)
+    logger.debug("Extra context: %s", extra_context)
+
+    try:
+        cookiecutter(
+            template_repository,
+            directory="cluster-template",
+            output_dir=str(output_path),
+            overwrite_if_exists=False,
+            no_input=True,
+            extra_context=extra_context,
+        )
```

Comment:
```suggestion
        cookiecutter(
            template_repository,
            checkout=template_version
            directory="cluster-template",
            output_dir=str(output_path),
            overwrite_if_exists=False,
            no_input=True,
            extra_context=extra_context,
        )
```

Right now when you use the phd_cluster_create command, it will always use the cookiecutter templates from the main branch. Should we use the `template_version` instead?

Reply:
👍